zyte_common_items/items.py

```diff
@@ -1202,3 +1206,64 @@ class SocialMediaPost(Item):
     metadata: Optional[SocialMediaPostMetadata] = attrs.field(
         default=None, converter=attrs.converters.optional(MetadataCaster(SocialMediaPostMetadata)), kw_only=True  # type: ignore
     )
+
+
+_TEMPLATE_ENVIRONMENT = jinja2.Environment()
+_TEMPLATE_ENVIRONMENT.filters["quote_plus"] = quote_plus
+
+
+def _render(string: str, **kwargs) -> str:
+    return _TEMPLATE_ENVIRONMENT.from_string(string).render(**kwargs)
+
+
+@attrs.define(kw_only=True)
+class SearchRequestTemplate:
+    """:ref:`Request template <request-templates>` to build a search
+    :class:`~zyte_common_items.Request`."""
+
+    #: :doc:`Jinja template <jinja:templates>` for :class:`Request.url
+    #: <zyte_common_items.Request.url>`.
+    url: str
+
+    #: :doc:`Jinja template <jinja:templates>` for :class:`Request.method
+    #: <zyte_common_items.Request.method>`.
+    method: str = "GET"
+
+    #: :doc:`Jinja template <jinja:templates>` for :class:`Request.body
+    #: <zyte_common_items.Request.body>`.
+    #:
+    #: It must be a plain :class:`str`, not :class:`bytes` or a Base64-encoded
+    #: :class:`str`. Base64-encoding is done by :meth:`request` after rendering
+    #: this value as a Jinja template.
+    #:
+    #: Defining a non-UTF-8 body is not supported.
+    body: Optional[str] = None
```

Comment:
Would this still work ok if we want a form-encoded body (e.g. a search form which makes a POST request), or it's too uncommon?

Reply:
Do you have any particular concern in mind? I assume it would be supported as long as the request body remained UTF-8, even though with a pure template (as opposed to [a future DSL](https://github.com/zytedata/zyte-common-items/pull/80#discussion_r1514159312) to simplify some common/tricky scenarios) it could be a bit cumbersome, and I wonder if we would need to expose some special variable to the template, like a random string of some length, to allow generating a different multi-part boundary every time (if hard-coding a boundary were to lower success rates).

Reply:
> Do you have any particular concern in mind? 

Mostly about ergonomics of constructing the correct payload, exactly the point you covered, thanks 👍